org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/factory/IQuickEntryFactory.java

```diff
@@ -0,0 +1,50 @@
+/******************************************************************************
+ * Copyright (C) Contributors                                                 *
+ * This program is free software; you can redistribute it and/or modify it    *
+ * under the terms version 2 of the GNU General Public License as published   *
+ * by the Free Software Foundation. This program is distributed in the hope   *
+ * that it will be useful, but WITHOUT ANY WARRANTY; without even the implied *
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.           *
+ * See the GNU General Public License for more details.                       *
+ * You should have received a copy of the GNU General Public License along    *
+ * with this program; if not, write to the Free Software Foundation, Inc.,    *
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.                     *
+ * Contributors:                                                              *
+ *   Andreas Sumerauer                                                        *
+ *****************************************************************************/
+
+package org.adempiere.webui.factory;
+
+import org.adempiere.webui.grid.AbstractWQuickEntry;
+
+/**
+ * 
+ * @author Andreas Sumerauer
+ *
+ */
+public interface IQuickEntryFactory {
+
+	/**
+	 * @param WindowNo
+	 * @param AD_Window_ID
+	 * @return new QuickEntry instance
+	 */
+	public default AbstractWQuickEntry newQuickEntryInstance(int WindowNo, int AD_Window_ID) {
+		return newQuickEntryInstance(WindowNo, 0, AD_Window_ID);
+	}
+	
+	/**
+	 * @param WindowNo
+	 * @param TabNo
+	 * @param AD_Window_ID
+	 * @return new QuickEntry instance
+	 */
+	public AbstractWQuickEntry newQuickEntryInstance(int WindowNo, int TabNo, int AD_Window_ID);
+	
+	/**
+	 * @param AD_Window_ID
+	 * @return new QuickEntry instance
+	 */
+	public AbstractWQuickEntry newQuickEntryInstance(int AD_Window_ID);
+	
+}
```

Comment:
what's the use case for the first and third method ? don't see it being use anywhere.

Reply:
The original WQuickEdit class has public constructors with the same signature. The motivation for the additional factory methods was just to provide the same interface. 
If they are in fact redundant I'll be happy to remove them.

Reply:
alright, that looks ok. checking the usage of the WQuickEntry constructors, your changes miss the use of WQuickEntry at InfoWindow.

Reply:
You are right. Looking at the codebase I'd say we should in fact only expose the interface that is used by WTableDirEditor. I'll remove the other factory functions then.

Reply:
hmm .. shouldn't the interface be used in InfoWindow as well ?

Reply:
Yes. I missed that completely. It's fixed now.